### DISCUSSION TITLE
How access an option picker as Variable

### DISCUSSION THREAD

**Discussion by immohacker** [Budibase/budibase#11060] [Help] [answered]
Hi, i would copy an value from an option picker to clipboard

My option Picker.

![Auswahl_1009](https://github.com/Budibase/budibase/assets/109019584/03b03d93-986d-4041-9f90-b9c7be7aed90)

My values:

![Auswahl_1010](https://github.com/Budibase/budibase/assets/109019584/d7880241-a96b-4024-8441-85dc73206b44)

The picker has the ID as value but shows the as label column the human readable value.
If i want try to access this i only get the id. Which is right, because the id is its value.
How can i access an option pickerts lbael column?

My embed button with code
![Auswahl_1011](https://github.com/Budibase/budibase/assets/109019584/e1c8c52c-78fb-43fc-8c26-b9c95286b521)

The code
![Auswahl_1012](https://github.com/Budibase/budibase/assets/109019584/42792c2f-ab44-4d80-b9e0-93d122b61bb9)

It returned the ID.
![Auswahl_1013](https://github.com/Budibase/budibase/assets/109019584/01b42bf2-6278-4090-bf0a-0d30a0a34c67)

How can i do it, that the label column would returned, or better the fields to the ID.
So that i can build an new string from the fileds. 

strasse + ort + plz

the field adressanzeige is the labelcolumn in the option picker
![Auswahl_1014](https://github.com/Budibase/budibase/assets/109019584/cced9025-3024-4084-a4ae-57c298c8cd92)

I hope i can describe my problem.
Thank you!
Best regards Bonkersdeluxe

**Comment by ConorWebb96** [comment]
Hey @immohacker,

I think I may have answered something similar to this recently. You can see the answer here; https://github.com/Budibase/budibase/discussions/10950 

As I said there it would be a good idea to create an enhancement request for this. As its the second time someone has wanted this recently.

**Comment by immohacker** [comment]
Hi ConorWebb96
Thank you for Answer.
i looked at your linked Answer.
But i see it right, he manually writes the fields with value and labels?

me is helped if i can create in javascript an query.
I have the id. eg select * from table where _id = {{ variable }}
Iam not so good in java script.

But it would be nice to have an feature the access label.
Where i can create an enhancement request?

Thank you!
Best regards Bonkersdeluxe

**Comment by immohacker** [comment]
Hi

I try an other solution.
Ive created an script but my script hangs at this line.

`const rows = $('dataRechnungsadressemeldeadresse.Rows');`

Here is the whole script. I know that the copy to clipboard copies nothing. 
But the function would not executed. because it stopped as the line above.

`<button name="rechnungsadressemeldeadresse" onclick="( function() 
{  
const rows = $('dataRechnungsadressemeldeadresse.Rows');
let street= '';
let ort = '';
let plz = '';

for (let i in rows) {
   if ('{{ Form.Fields.rechnungsadressemeldeadressepicker }}' == rows[i]._id)
    {
            street = rows[i].strasse;
            ort = rows[i].ort;
            plz = rows[i].plz;
    }
}

navigator.clipboard.writeText('').then(function (){alert ( '  Copied!')}) }
)()">Copy Link</button>`

Ive marked the Data Provider. It exist
![Auswahl_1015](https://github.com/Budibase/budibase/assets/109019584/00c25e18-f7b7-4cff-922b-ca9685e0f072)

What make  i wrong?
Thank you!
Best regards Bonkersdeluxe

**Comment by immohacker** [comment, marked as answer]
Hi, so i  got it:

With an javascript thats build the button:

const rows = $("dataRechnungsadressemeldeadresse.Rows");
let street = "strasse";
let ort = "ort";
let plz = "plz";
for (let i in rows) {
   if ($("Form.Fields.rechnungsadressemeldeadressepicker") == rows[i]._id)
    {
            street = rows[i].strasse;
            ort = rows[i].ort;
            plz = rows[i].plz;
    }
}
let text = street +" "+ plz +" " + " " + ort;
let button = `<button name="rechnungsadressemeldeadresse" onclick="( function() 
{  
    navigator.clipboard.writeText('${text}').then(function (){alert ('${text}  Copied!')}) }
)()">Copy Link</button>`;
return button;

![Auswahl_1016](https://github.com/Budibase/budibase/assets/109019584/e19c987e-c473-43ad-a331-530e7b502e1f)

The second benefit is, so i can access all fields in the Dataset and not label only

But it would be nice to get this thing easier, if only label is needed
Where i must create an enhancement?

Thank you!
Best regards Bonkersdeluxe
